tests/red-black-tests.lisp

```diff
@@ -36,9 +36,11 @@
           (if (== left-ct right-ct)
               (Ok (+ left-ct (match c
                                ((red-black/tree::Black) 1)
-                               ((red-black/tree::Red) 0))))
+                               ((red-black/tree::Red) 0)
+                               (_ (error "unreachable")))))
               (Err (DifferentCountToBlack left-ct left
-                                          right-ct right)))))))))
+                                          right-ct right))))))
+      (_ (error "unreachable")))))
```

Comment:
These two should probably have a dedicated variant in `InvariantError`. Something like:

```
(define-type (InvariantError :elt)
  ;;  other variants
  (IllegalColor (red-black/tree:Tree :elt)))
```

And then have the two branches, instead of calling `error`, be:

```
(if (== (left-ct right-ct))
  (match c
    ((red-black/tree::Black) (Ok (1+ left-ct)))
    ((red-black/tree::Red) (Ok left-ct))
    (_ (Err (IllegalColor tre))))
  (Err (DifferentCountToBlack left-ct left right-ct right)))
```

```
((DoubleBlackEmpty) (Err (IllegalColor tre)))
```